plugins/backend/catalog-backend-module-aws/src/providers/AWSECRRepositoryEntityProvider.test.ts

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright 2025 Larder Software Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { STS, GetCallerIdentityCommand } from '@aws-sdk/client-sts';
+import {
+  ECRClient,
+  DescribeRepositoriesCommand,
+  ListTagsForResourceCommand,
+  DescribeRepositoriesCommandOutput,
+  ListTagsForResourceCommandOutput,
+} from '@aws-sdk/client-ecr';
+import { mockClient } from 'aws-sdk-client-mock';
+import { createLogger, transports } from 'winston';
+import { ConfigReader } from '@backstage/config';
+import { EntityProviderConnection } from '@backstage/plugin-catalog-backend';
+import { AWSECRRepositoryEntityProvider } from './AWSECRRepositoryEntityProvider';
+
+// @ts-ignore
```

Comment:
I feel bad doing this.. but I was really struggling with getting these types in line, and the typescript errors only appear in the tests. The runtime code types are fine.